inc/commonglpi.class.php

```diff
@@ -999,7 +1008,11 @@ class='navicon left'>
          $input   = $ma->getInput();
 
          if ($this->isEntityAssign()) {
-            $input['entity_restrict'] = $this->fields['entities_id'];
+            if (is_subclass_of($this, 'CommonDBChild')) {
```

Comment:
Ths `getEntityID()` seems to handle this more precisely. It is implemented in `CommonDBTM` and overrided in `CommonDBChild`.